app/mci/cmd/create.go

```diff
@@ -173,25 +177,34 @@ func getClusters(kubeconfig string) ([]string, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error in getting contexts from kubeconfig: %s", err)
 	}
+	glog.Infof("contexts found:\n%v", output)
 	return strings.Split(output, "\n"), nil
 }
 
 // Creates the given ingress in the given list of clusters.
 func createIngressInClusters(kubeconfig, ingressFilename string, clusters []string) error {
-	kubectlArgs := []string{"kubectl"}
-	if kubeconfig != "" {
-		kubectlArgs = append(kubectlArgs, fmt.Sprintf("--kubeconfig=%s", kubeconfig))
+	var ing v1beta1.Ingress
+	if err := unmarshall(ingressFilename, &ing); err != nil {
+		return fmt.Errorf("error in unmarshalling the yaml file %s, err: %s", ingressFilename, err)
 	}
+	glog.V(2).Infof("Unmarshaled this ingress:\n%+v", ing)
+
 	// TODO(nikhiljindal): Validate and optionally add the gce-multi-cluster class annotation to the ingress YAML spec.
-	createArgs := append(kubectlArgs, []string{"create", fmt.Sprintf("--filename=%s", ingressFilename)}...)
 	for _, c := range clusters {
-		fmt.Println("Creating ingress in context:", c)
-		contextArgs := append(createArgs, fmt.Sprintf("--context=%s", c))
-		output, err := runCommand(contextArgs)
+		fmt.Println("\nHandling context:", c)
+		client, err := getClientset(kubeconfig, c)
+		if err != nil {
+			glog.Errorf("Error getting kubectl client interface: %v", err)
+			return err
+		}
+		fmt.Println("Using this namespace for ingress:", ing.Namespace)
+		ingInterface := client.Extensions().Ingresses(ing.Namespace)
```

Comment:
ingInterface var not required
`actualIng, err := client.Extensions().Ingresses(ing.Namespace).Create(&ing)`